paasta_tools/cli/cmds/validate.py

```diff
@@ -627,30 +646,30 @@ def _validate_active_requests_autoscaling_configs(
 
 
 def _validate_arbitrary_promql_autoscaling_configs(
-    autoscaling_params: AutoscalingParamsDict,
-):
+    metrics_provider_config: MetricsProviderDict,
+) -> None:
     # This is a bit incorrect, since the default autoscaling setpoint is 0.8,
     # someone could theoretically bypass this by setting setpoint: 0.8 in their
     # soaconfigs; but I think it's approximately fine for now
     if (
-        autoscaling_params.get("setpoint", DEFAULT_AUTOSCALING_SETPOINT)
+        metrics_provider_config.get("setpoint", DEFAULT_AUTOSCALING_SETPOINT)
         != DEFAULT_AUTOSCALING_SETPOINT
     ):
         raise AutoscalingValidationError(
             "setpoint is not supported for arbitrary PromQL"
         )
-    if not autoscaling_params.get("prometheus_adapter_config"):
+    if not metrics_provider_config.get("prometheus_adapter_config"):
         raise AutoscalingValidationError(
             "arbitrary promql metrics provider requires prometheus_adapter_config to be set"
         )
 
 
-def _validate_autoscaling_config(autoscaling_params: AutoscalingParamsDict):
-    metrics_provider = autoscaling_params["metrics_provider"]
-    for field in INVALID_AUTOSCALING_FIELDS[metrics_provider]:
-        if field in autoscaling_params:
+def _validate_autoscaling_config(metrics_provider_config: MetricsProviderDict):
```

Comment:
```suggestion
def _validate_autoscaling_config(metrics_provider_config: MetricsProviderDict) -> None:
```

i was debating not commenting here, but then i saw that you fixed the missing typing above :p

Reply:
lol I'm bad at saying things return nothing apparently